### PR TITLE
For #2818 - Adds Picture-in-Picture feature

### PR DESCRIPTION
**Pull request by ekager**
Needs https://github.com/mozilla-mobile/android-components/pull/6557 to land to work properly in custom tabs

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by ekager** [comment]
The required AC changes have landed so this is ready for review now!

**Comment by ekager** [comment]
@jonalmeida do you want to review this? You probably have most context :)

File: app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt [hunk 1]
```diff
@@ -813,6 +791,63 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Session
...
+        if (pipFeature.onHomePressed()) {
+            return true
+        }
+        return false
+    }
+
+    private fun pipModeChanged(enabled: Boolean) {
+        val sessionManager = requireComponents.core.sessionManager
+        val customTabSession = customTabSessionId?.let { sessionManager.findSessionById(it) }
+        val session = customTabSession ?: sessionManager.selectedSession
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt:804]: We could use our working SessionManager ktx here as well! 😄

File: app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt [hunk 2]
```diff
@@ -324,6 +326,13 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Session
                 view = view
             )
 
+            pipFeature = PictureInPictureFeature(
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt:329]: Did we want to consider a nightly-only feature flag?

Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt]: good call at least for a bit

File: app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt (hunk 1 above)
Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt]: 👍

**Pull request merged**